util/Migrator/DbScripts/2024-01-16_00_AddUnassignedCiphersQuery.sql

```diff
@@ -0,0 +1,31 @@
+CREATE OR ALTER PROCEDURE [dbo].[CipherOrganizationDetails_ReadUnassignedByOrganizationId]
+    @OrganizationId UNIQUEIDENTIFIER
+AS
+BEGIN
+    SET NOCOUNT ON
+
```

Comment:
It would be more efficient to not use the not in clause and do where:

```

SELECT
    C.*,
    CASE
        WHEN O.[UseTotp] = 1 THEN 1
        ELSE 0
    END [OrganizationUseTotp]
FROM
    [dbo].[CipherView] C
LEFT JOIN
    [dbo].[OrganizationView] O ON O.[Id] = C.[OrganizationId]
LEFT JOIN
    [dbo].[CollectionCipher] CC ON C.[Id] = CC.[CipherId]
LEFT JOIN
    [dbo].[Collection] S ON S.[Id] = CC.[CollectionId]
    AND S.[OrganizationId] = C.[OrganizationId]
WHERE
    C.[UserId] IS NULL
    AND C.[OrganizationId] = @OrganizationId
    AND CC.[CipherId] IS NULL
```

Reply:
Thanks for the suggestion! Updated in [db7da53](https://github.com/bitwarden/server/pull/3676/commits/db7da53a99292dc3b8560d4aa9945ab4661258f5)